waku/v2/peermanager/service_slot.go

```diff
@@ -0,0 +1,75 @@
+package peermanager
+
+import (
+	"sync"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+	"github.com/waku-org/go-waku/waku/v2/utils"
+)
+
+type peerMap struct {
```

Comment:
Any specific reason for choosing a map over an array here? 
Having an array makes it easy to choose a peer randomly. With a map selecting at random would be difficult.